tensorflow/python/framework/tensor_util_test.py

```diff
@@ -662,6 +663,69 @@ def testShapeEquals(self):
     self.assertFalse(tensor_util.ShapeEquals(t, [1, 4]))
     self.assertFalse(tensor_util.ShapeEquals(t, [4]))
 
+  def testMockArray(self):
+    class MockArray(object):
+      def __init__(self, array):
+        self.array = array
+
+      def __array__(self, dtype=None):
+        return np.asarray(self.array, dtype)
+
+    ma = MockArray(np.array([10, 20, 30]))
```

Comment:
Can you make these arrays smaller (maybe single-element) so the proto will fit in the test?